blockchain/src/pretty.rs

```diff
@@ -0,0 +1,405 @@
+use std;
+use std::fmt;
+use std::string::String;
+
+use ansi_term::Colour;
+
+use block;
+use genesis;
+use normal;
+use types;
+use wallet_crypto;
+
+static DISPLAY_INDENT_SIZE: usize = 4; // spaces
+static DISPLAY_INDENT_LEVEL: usize = 0; // beginning starts at zero
+static DISPLAY_USE_COLOR: bool = false; // no color for display implementations
+
+type Assoc = (String, Val);
+
+// core pretty-printer type which facilitates homogenous ordered key-value pairs
+pub enum Val {
+    Single(Box<fmt::Display>),
```

Comment:
one could use [`fmt::Argument`](https://doc.rust-lang.org/std/fmt/struct.Arguments.html) here and remove the `Box` This way we are not link to `fmt::Display` which has a specific meaning in rust, different from the pretty printing we want to use.

A good example of usage is the crate [log](https://github.com/rust-lang-nursery/log) which uses the `fmt::Argument` structure instead of the `fmt::Display` trait so the user can use any formatting trait (Debug, hexa...).

Reply:
if i switch to using the `fmt.Arguments` struct, then would i be using the `format_args!` macro every time i construct a `Single`?

Reply:
yes that would be the idea.

Reply:
I'm not entirely sure how the log package you linked me to works, but [i'm encountering an error about a dropped borrowed value](http://play.rust-lang.org/?gist=d3e687087ab418cdda66e4ed14cdf78a&version=undefined&mode=undefined) whenever I attempt to save the result of `format_args!`. If I'm missing something, please let me know. However, in light of that I'll probably use `String` for the plain-value case.